test/adversary/adversary-code/block-assembler.mjs

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-await-in-loop, no-unused-vars */
+
+/**
+This module does all of the heaving lifting for a Proposer: It assembles blocks
+from posted transactions and proposes these blocks.
+*/
+import config from 'config';
+import logger from 'common-files/utils/logger.mjs';
+import { getMostProfitableTransactions } from './database.mjs';
+import Block from './block.mjs';
+
+const { TRANSACTIONS_PER_BLOCK } = config;
+
+let count = 0;
+async function makeBlock(proposer, number = TRANSACTIONS_PER_BLOCK) {
+  logger.debug('Block Assembler - about to make a new block');
+  // pick a random number between 0 to 6 (error length - 1)
+  const errorIndex = Math.floor(Math.random() * 0);
```

Comment:
Not sure what `errorIndex` does here - it seems to always be 0. `count` is always 0 as well.

Does this mean that `getMostProfitableTransactions` always makes valid transactions?

Reply:
It's a global variable. So it is incremented with each block proposed.
Non valid blocks get created as it increments. 
This is a test run from github action
<img width="1285" alt="Screenshot 2022-03-07 at 09 27 57" src="https://user-images.githubusercontent.com/36898048/157004271-505e96e5-d67a-4904-9c06-d9cb304c8827.png">

Reply:
ah ok - so `errorIndex` is always zero because we are not doing the random thing for now